Assets/_BForBoss/_Core/Scripts/Managers/BaseWorldManager.cs

```diff
@@ -79,26 +83,27 @@ protected virtual void Awake()
             _inputSystem.OnPausePressed += HandlePausePressed;
             _stateManager.OnStateChanged += HandleStateChange;
             _environmentManager = gameObject.AddComponent<EnvironmentManager>();
-            SceneManager.LoadScene("AdditiveWeaponManager", LoadSceneMode.Additive);
-            SceneManager.LoadScene("AdditiveUserInterfaceScene", LoadSceneMode.Additive);
+            SceneManager.LoadScene(ADDITIVE_WEAPON_SCENE_NAME, LoadSceneMode.Additive);
+            SceneManager.LoadScene(ADDITIVE_USERINTERFACE_SCENE_NAME, LoadSceneMode.Additive);
 #if (UNITY_EDITOR || DEVELOPMENT_BUILD)
-            SceneManager.LoadScene("AdditiveDebugScene", LoadSceneMode.Additive);
+            SceneManager.LoadScene(ADDITIVE_DEBUG_SCENE_NAME, LoadSceneMode.Additive);
 #endif
+            
+            SceneManager.sceneLoaded += OnAdditiveSceneLoaded;
```

Comment:
Shouldn't this be added before the SceneManager.LoadScene calls? Just in case a scene gets loaded before this statement is reached (although realistically not possible I guess)